experiments/krico/workloads/utils.go

```diff
@@ -0,0 +1,109 @@
+package workload
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/intelsdi-x/swan/pkg/conf"
+	"github.com/intelsdi-x/swan/pkg/executor"
+	"github.com/intelsdi-x/swan/pkg/experiment"
+	"github.com/libvirt/libvirt-go"
+	"github.com/pkg/errors"
+)
+
+const (
+	// TypeCaching is constant name of caching workload.
+	TypeCaching = "caching"
+)
+
+var (
+	snapStartCommand = "systemctl restart snap-telemetry"
+	aggressorAddress = conf.NewStringFlag("aggressor_address", "IP address of aggressor node.", "127.0.0.0")
+)
+
+// RunCollectingMetrics runs metric gathering experiment for each type of workload.
+func RunCollectingMetrics(experimentID string) {
+	CollectingMetricsForCachingWorkload(experimentID)
+}
+
+// RunWorkloadsClassification runs classification experiment for each type of workload. Return instances id.
+func RunWorkloadsClassification(experimentID string) []string {
+	var instances []string
+
+	instances = append(instances, ClassifyCachingWorkload(experimentID))
+
+	return instances
+}
+
+// StartSnapService starts Snap Telemetry Framework.
+func StartSnapService(address string) error {
+
+	startSnapExecutor, err := executor.NewRemoteFromIP(address)
+	if err != nil {
+		err = errors.Wrapf(err, "Cannot obtain Snap-telemetry executor on %q !")
+		return err
+	}
+
+	startSnapTaskHandle, err := startSnapExecutor.Execute(snapStartCommand)
+	if err != nil {
+		err = errors.Wrapf(err, "Cannot execute %q command on %q !", snapStartCommand, address)
+		return err
+	}
+
+	//	Wait until start
+	_, err = startSnapTaskHandle.Wait(0)
+	if err != nil {
+		err = errors.Wrap(err, "Cannot start Snap-telemetry service !")
+		return err
+	}
+
+	return nil
+}
+
+// GetInstanceCgroup provides cgroup of libvirt instance.
+func GetInstanceCgroup(hypervisorInstanceName string, hypervisorAddress string) (string, error) {
+
+	conn, err := libvirt.NewConnectReadOnly("qemu+ssh://root@" + hypervisorAddress + "/system")
+	if err != nil {
+		return "", fmt.Errorf("couldn't connect to libvirt: %v", err)
+	}
+	defer conn.Close()
+
+	domain, err := conn.LookupDomainByName(hypervisorInstanceName)
+	if err != nil {
+		return "", fmt.Errorf("couldn't get instance domain: %v", err)
+	}
+
+	instanceID, err := domain.GetID()
+	if err != nil {
+		return "", fmt.Errorf("couldn't get instance domain id: %v", err)
+	}
+
+	instanceName := strings.Replace(hypervisorInstanceName, "-", `\x2d`, -1)
```

Comment:
Replace magic string with a constant.

Reply:
Could you said what you mean, more precisely?